src/mscorlib/shared/System/ReadOnlyMemory.cs

```diff
@@ -125,11 +127,11 @@ internal ReadOnlyMemory(OwnedMemory<T> owner, int index, int length)
         public ReadOnlyMemory<T> Slice(int start)
         {
             if ((uint)start > (uint)_length)
-                ThrowHelper.ThrowArgumentOutOfRangeException();
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.start);
+            }
 
-            if (_index < 0)
-                return new ReadOnlyMemory<T>((OwnedMemory<T>)_arrayOrOwnedMemory, (_index & RemoveOwnedFlagBitMask) + start, _length - start);
-            return new ReadOnlyMemory<T>((T[])_arrayOrOwnedMemory, _index + start, _length - start);
+            return new ReadOnlyMemory<T>(_object, _index + start, _length - start);
```

Comment:
The same optimizations should be done for regular `Memory<T>` slicing as well.

Reply:
Yes, some of them can be.  Let's do it as a separate change.

Reply:
Actually, I'll just take care of it as part of this change, it's easy enough.